text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+use them via strongly-typed classes. IDEs will show type information and inline help derived from the
+extension schema.
+
+```ts
+import { Cluster } from '@cdk-cloudformation-extensions/awsqs-eks';
+
+new Cluster(this, 'MyEksCluster', {
+  name: 'my-new-cluster',
+  tags: [ { key: 'foo', value: 'bar' } ],
+});
+```
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We are doing this because CDK users would like to be able to use CloudFormation extensions 
+(modules, resources) as first-class citizens in CDK code. Extensions and native resource types
+should look and feel the same way when defined in CDK apps.
+
+Additionally, we would like to surface CloudFormation extensions in the upcoming Construct Hub
+and to that end, if we simply publish a construct library that includes constructs for all the public,
+extensions, this library will naturally be discoverable through the Hub.
+
+### Why should we _not_ do this?
+
+We have a longer term plan ([RFC#77](https://github.com/aws/aws-cdk-rfcs/issues/77)) to add 
+support for a CLI command called `cdk import` which will allow users to generate L1s from any
+registry type just-in-type and add them to their project (this is similar to CDK8s and CDKtf).
+
+### What is the technical solution (design) of this feature?
+
+The idea is to do something similar to what we do with the CloudFormation L1s 
+in the AWS CDK and simply publish JSII construct libraries which includes 
+statically-generated L1s for all the public registry extensions.
+
+#### Library per type
+
+CloudFormation extensions are semantically versioned. This means that new versions of an extension
+may include breaking changes. We also want to allow users to pick up a specific version of an extension.
+In light of these constraints, we determined that the best path forward is to publish each extension
+as a separate module, with a version that corresponds to the extension version in the CloudFormation
+registry. This way, we will have 1:1 alignment and users will have the most freedom.
+
+#### Code Generator
+
+The code generator will be executed daily (e.g. through a GitHub workflow),
+query the CloudFormation Registry, and generate L1s for all the types based
+on their metadata and schema.
+
+We can use the `ListTypes` API to list all the types in the registry:
+
+```shell
+aws cloudformation list-types --visibility=PUBLIC
+```
+
+And then for each type, we can use `DescribeType` to retrieve the type information and its schema. 
+
+For example, this command will return the description of the `AWSQS::EKS::Cluster` resource:
+
+```shell
+aws cloudformation describe-type --arn arn:aws:cloudformation:us-east-1::type/resource/408988dff9e863704bcc72e7e13f8d645cee8311/AWSQS-EKS-Cluster
+```
+
+For reference: the output of this command can be found [here](https://gist.github.com/eladb/bf417c07444027d6954360295df4ee37#file-awsqs-vpc-vpcqs-module-json). 
```

Comment:
(minor)

Generally, avoid using gist. It's lifecycle and ownership model is different from RFC.
Prefer using appendices instead.

The output doesn't look too big to add these inline.

Reply:
I think a gist is a good tool for this. I understand what you are saying about the lifecycle, but this is just a raw dump of an output of a command (300 lines), so I'll keep these in a gist. Seems to me like a pragmatic thing to do.